Sources/ProjectDescription/Headers.swift

```diff
@@ -82,4 +88,48 @@ public struct Headers: Codable, Equatable {
             exclusionRule: exclusionRule
         )
     }
+
+    public static func headers(from list: FileList,
+                               umbrella: Path,
+                               private: FileList? = nil,
+                               allOthersAsProject: Bool) -> Headers
+    {
+        return .init(
+            public: list,
+            umbrellaHeader: umbrella,
+            private: `private`,
+            project: allOthersAsProject ? list : nil,
+            exclusionRule: .projectExcludesPrivateAndPublic
+        )
+    }
+
+    /// all left found headers from the list
+    /// (after public/private scopes loading)
+    /// will be added with `project` visibility
+    public static func allHeaders(from list: FileList,
```

Comment:
What's the use case for this method?

Reply:
I used it from discussion about this request: https://github.com/tuist/tuist/discussions/3784
The main idea: "I want all headers at `DIR`, mark as `public` all from the umbrella, mark as `private` from the list, and all other found - as `project`